### DISCUSSION TITLE
Allowing companies to submit data directly

### DISCUSSION THREAD

**Discussion by tomsoderlund** [Open-Sustainability-Index/open-sustainability-index-backend#14] [Ideas]
We have a goal to allow companies to submit data directly (with review), and to stop using Google Sheets.

**Success:** 100 new companies in a month (500 as stretch goal)

Thoughts on the quickest way there:

- [x] Supabase/Postgres becomes _source-of-truth_ (before: Google Sheets)
- [x] Supabase will be the new admin interface
- [ ] New “Report data” form that feeds straight into Supabase: https://preview--sustainability-reporting-hub.gptengineer.run
  - [x] “Company name” will be a dropdown where also new companies can be created
  - [x] Easier to input multiple years (as columns in form?)
  - [x] Reporter’s name and email (optional)
- [ ] New rows (in tables `company`, `emission`, `target`, `commitment`) needs to be marked as `status` "To be reviewed" - **what statuses needed**?
- [ ] To publish the submitted data, OSI admins can change `status` in Supabase. This will be a bit awkward since data is found in 4 different tables, difficult to get a good overview. See a v2 below*.
- [ ] Site updates should go live without manual intervention, using Next.js incremental static regeneration (ISR)

v1.1

- [x] **AI image upload**
- [ ] Separate image upload for Revenue

*v2:

- [ ] Custom UI for review process

Some questions:

- How to handle misspelled company names, merging of duplicate companies and their emissions/targets?

**Comment by achaido** [comment]
Great stuff! Question: make the reporting page identical to the [public company pages](https://www.opensustainabilityindex.org/company/3m), just with input fields instead incl. a submit button. The same view is then used as admin interface. That way we can reuse the page we already have, both for reporting and reviewing. Would that make sense, and if so, would it also be in line with "quickest way there"?

**Reply by tomsoderlund** [reply]
Agree 100%, makes sense
